jarbas/frontend/elm/Documents/Inputs/View.elm

```diff
@@ -10,6 +10,42 @@ import Material.Textfield as Textfield
 import Material.Typography as Typography
 import Documents.Inputs.Model exposing (Model, Field, model)
 import Documents.Inputs.Update exposing (Msg(..), update)
+import Internationalization exposing (TranslationId(..), translate)
+import Regex
+
+
+{-| Matches a date un the YYYY-MM-DD format
+
+    >>> matchDate "1970-01-01"
+    True
+
+    >>> matchDate "foo-42-01"
+    False
+
+-}
+matchDate : String -> Bool
+matchDate value =
+    if String.isEmpty value then
+        True
+    else
+        let
+            regex : Regex.Regex
+            regex =
+                Regex.regex
+                    "[\\d]{4}-[\\d]{2}-[\\d]{2}"
```

Comment:
It does look weird having to escape the slashes, bikeshedding a bit, that gave me the idea: what if we had a lib for building type-safe regexes? Turns out that there is already one! http://package.elm-lang.org/packages/ydschneider/regex-builder/1.0.0/RegexBuilder

Probably overkill and maybe not very friendly for other developers as one of the advantages of regexes is that they are mostly universal across languages.